clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -425,6 +427,18 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
         Attrs.insert(A);
       else if (auto *A = FD->getAttr<ReqdWorkGroupSizeAttr>())
         Attrs.insert(A);
+      else if (auto *A = FD->getAttr<SYCLIntelKernelArgsRestrictAttr>()) {
+        // Allow the intel::kernel_args_restrict only on the lambda (functor
```

Comment:
Please do not use "functor" word.
https://github.com/intel/llvm/pull/654#issuecomment-533545781

Reply:
Done.